app/controllers/users_controller.rb

```diff
@@ -30,13 +30,25 @@ def index
 
   def show; end
 
+  def edit
+    redirect_to user_path(@user) unless @user.active?
+  end
+
   def update
     if @user.update(user_params)
       if @user == current_user
         bypass_sign_in @user
         flash[:notice] = I18n.t("devise.passwords.updated") if user_params.key?("password")
         redirect_to account_path
       else
+        case user_params[:active]
+        when "false"
+          @user.update!(confirmed_at: nil, sign_in_count: 0, encrypted_password: "")
```

Comment:
Why do we set the password to empty string?

Reply:
So that after getting reactivated the user wouldn't be able to login with the old creds

Reply:
I mean if it's not required to reset the password, I'd happily remove that bit. Idk why I assumed it was